package/MDAnalysis/analysis/base.py

```diff
@@ -220,6 +416,66 @@ def __setstate__(self, state):
         self.data = state
 
 
+class ResultsGroup:
+    """Simple class that does aggregation of the results from independent remote workers in AnalysisBase
+    """
+    def __init__(self, lookup: dict[str, Callable] = None):
+        """Initializes the class, saving aggregation functions in _lookup attribute.
+
+        Parameters
+        ----------
+        lookup : dict[str, Callable], optional
+            aggregation functions lookup dict, by default None
+        """
+        self._lookup = lookup
+
+    def merge(self, objects: Sequence[Results], do_raise: bool = True) -> Results:
+        """Merge results into a single object.
+        If objects contain single element, returns it while ignoring _lookup attribute.
+
+        Parameters
+        ----------
+        do_raise : bool, optional
+            if you want to raise an exception when no aggregation function for a particular argument is found, by default True
+
+        Returns
+        -------
+        Results
+            merged Result object
+
+        Raises
+        ------
+        ValueError
+            if no aggregation function for a key is found and do_raise = True
+        """
+        if len(objects) == 1:
+            rv = objects[0]
+        else:
+            rv = Results()
+            for key in objects[0].keys():
+                agg_function = self._lookup.get(key, None)
+                if agg_function is None and do_raise:
+                    raise ValueError(f"No aggregation function for {key=}")
+                results_of_t = [obj[key] for obj in objects]
+                rv[key] = agg_function(results_of_t)
+        return rv
+
+    @staticmethod
+    def flatten_sequence(arrs: list[list]):
+        return [item for sublist in arrs for item in sublist]
+
+    @staticmethod
+    def ndarray_sum(arrs: np.ndarray):
+        return np.array(arrs).sum(axis=0)
+
+    @staticmethod
+    def ndarray_mean(arrs: np.ndarray):
+        return np.array(arrs).sum(axis=0)
+
+    @staticmethod
+    def float_mean(floats: list[float]):
+        return sum(floats) / len(floats)
```

Comment:
just use
```suggestion
        return np.mean(floats)
```
This will make results consistent between `float_mean()` and `ndarray_mean()`.

Reply:
Fixed.